firestore/firestore_snippets/listen_changes.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// [START fs_listen_changes]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/firestore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// listenChanges listens to a query, returning the list of document changes.
+func listenChanges(w io.Writer, projectID string) error {
+	// projectID := "project-id"
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Let's pass a `context.Context` in so we can do a shorter timeout for tests. Include this 30 second timeout here so users don't end up with an infinite loop.